### PR TITLE
✨ Add n_observations to tiledbsoma artifacts

### PR DESCRIPTION
**Pull request by Koncopd**
Fixes https://github.com/laminlabs/lamindb/issues/1919

### CODE REVIEW

**Comment by sunnyosun** [comment]
Could we also populate the `accessor` for these artifacts so it's not "unknown" here?
Screenshot 2024-09-17 at 11 49 25

**Comment by Koncopd** [comment]
Yep

**Comment by Koncopd** [comment]
How should i call it? `tiledbsoma`?

**Comment by sunnyosun** [comment]
We currently have "AnnData", "DataFrame", "MuData", so maybe "TileDB-SOMA"?

**Comment by Koncopd** [comment]
Yes, makes sense, added like this.

**Comment by sunnyosun** [comment]
Ha, you did the same already! :D

File: lamindb/core/storage/_tiledbsoma.py [hunk 1]
```diff
@@ -186,11 +192,15 @@ def save_tiledbsoma_experiment(
...
         storepath,
         key=key,
         description=description,
         run=run,
         revises=revises,
         _is_internal_call=True,
-    ).save()
+    )
+    artifact.n_observations = n_observations
+    artifact._accessor = "TileDB-SOMA"
```
Comment (falexwolf) [inline lamindb/core/storage/_tiledbsoma.py:204]: Can we please keep it simple here and use `tiledbsoma` instead?

Comment (falexwolf) [reply lamindb/core/storage/_tiledbsoma.py:204]: I'll make the change.

It's mostly for aesthetical reasons, but `TileDB-SOMA` also isn't valid Python code whereas `tiledbsoma` is.

**Pull request merged**